source/common/common/assert.h

```diff
@@ -13,6 +16,46 @@ class ActionRegistration {
 };
 using ActionRegistrationPtr = std::unique_ptr<ActionRegistration>;
 
+/*
+ * EnvoyBugStackTrace captures and writes the stack trace to Envoy Bug
+ * to assist with getting additional context for reports
+ */
+class EnvoyBugStackTrace : Logger::Loggable<Logger::Id::envoy_bug> {
+public:
+  EnvoyBugStackTrace() = default;
+  /*
+   * Capture the stack trace
+   * Skip count is one as to skip the last call which is capture()
+   */
+  void capture() {
+    stack_depth_ = absl::GetStackTrace(stack_trace_, MaxStackDepth, /* skip_count = */ 1);
+  }
+
+  /*
+   * Logs each row of the captured stack into the envoy_bug log
+   */
+  void logStackTrace() {
+    ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::envoy_bug), error,
+                        "stacktrace for envoy bug");
+    for (int i = 0; i < stack_depth_; ++i) {
+      char out[1024];
+      const bool success = absl::Symbolize(stack_trace_[i], out, sizeof(out));
+      if (success) {
+        ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::envoy_bug), error,
```

Comment:
You're already using the mixin above, you can just use `ENVOY_LOG(error, "msg", ...)` here and elsewhere

Reply:
Done